taskmanager/src/Application/Report/Criteria.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types=1);
+
+namespace App\Application\Report;
+
+class Criteria
```

Comment:
I'm not sure whether criteria should be VO in such form. `Criteria` object accepts only status of the task. IMHO it's not enough even at the beginning. How to handle a case e.g. `createdAt BETWEEN(2017-01-01, 2017-02-02)` ?

Reply:
Add function can get name and object of single criteria element like bellow:
```php
public function add(string $name, CriteriaElementInterface $criteriaElement);
```